.github/workflows/android.yml

```diff
@@ -0,0 +1,79 @@
+name: Android Builds
+
+on: 
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+env:
+  CCACHE_DIR: ${{ github.workspace }}/ccache_dir
+
+jobs:
+  build:
+    name: android-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.architecture }}-${{ matrix.python_version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        architecture: ["x64",]
+        python_version: [3.7]
+        include:
+        - os: ubuntu-latest
+          architecture: "x64"
+        - os: macos-latest
+          architecture: "x64"
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Set env variables for subsequent steps (all)
+        run: |
+          echo "::set-env name=PIP_REQUIREMENTS_FILE::external/pip_requirements.txt"
+          echo "::set-env name=MATRIX_UNIQUE_NAME::${{ matrix.os }}-${{ matrix.architecture }}-${{ matrix.python_version }}"
+      
+      - name: Install system level applications (mac)
+        if: startsWith(matrix.os, 'macos')
+        id: macos-brew-install
+        run: |
+          brew install ccache
+          echo "::set-env name=CCACHE_INSTALLED::1"
+      
+      - name: Install system level applications (linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        id: ubuntu-apt-install
+        run: |
+          sudo apt install ccache
+          echo "::set-env name=CCACHE_INSTALLED::1"
+
+      - name: Cache ccache files
+        id: cache_ccache
+        uses: actions/cache@v2
+        with:
+          path: ccache_dir
+          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
```

Comment:
I think we should just remove python_version from the matrix entirely.

Reply:
Done.

Reply:
I think its good to have a Python version specified because of multiple reasons,
- Latest install_preqreqs and build scripts are using a feature of `subprocess.run` that is only available in Python 3.7+
- Github runners have Python 3.6 installed by default 
- Installing a fresh python environment starts us from a clean slate and is a better way of verifying our work

So yeah, we could remove it from the matrix but we should leave the "Setup Python" step in the workflow.